File: 1133_largest_unique.go
// Returns a counter of all elements in a integer array.
func counter(A []int) map[int]int {
    c := make(map[int]int)
    for i := 0; i < len(A); i++ {
        curr := A[i]
        _, ok := c[curr]
        if ok {
            c[curr]++
        } else {
            c[curr] = 1
        }
    }
    return c
}

// Given an array of integers arr, return the largest integer that only
// occurs once.
// If no integer occurs once, return -1.
func largestUniqueNumber(arr []int) int {
    c := counter(arr)
    max_uniq := -1
    for k, v := range(c) {
        if v == 1 && k > max_uniq {
            max_uniq = k
        }     
    }
    return max_uniq
}

func main {
	arr := []int{5,7,3,9,4,9,8,3,1}
    fmt.Println(largestUniqueNumber(arr))
}

File: 1133_largest_unique.py
from collections import Counter
from typing import List


def largest_unique_number(arr: List[int]) -> int:
    """
    Given an array of integers arr, return the largest integer that only
    occurs once.
    If no integer occurs once, return -1.
    """
    return max([k for k, v in Counter(arr).items() if v == 1], default=-1)


arr = [5, 7, 3, 9, 4, 9, 8, 3, 1]
print(largest_unique_number(arr))  # 8

arr2 = [9, 9, 8, 8]
print(largest_unique_number(arr2))  # -1
